File: 7-kyu/fizz-buzz-cuckoo-clock/swift/solution.swift
func fizzBuzzCuckooClock(_ time: String) -> String {
    let timeComponents = time.components(separatedBy: ":")
    let hours = Int(timeComponents[0])!
    let minutes = Int(timeComponents[1])!
    if minutes == 0 {
        return Array(repeating: "Cuckoo", count: (hours % 12 == 0 ? 12 : hours > 12 ? hours - 12 : hours)).joined(separator: " ")
    }
    else if minutes == 30 {
        return "Cuckoo"
    }
    else if minutes % 15 == 0 {
        return "Fizz Buzz"
    }
    else if minutes % 5 == 0 {
        return "Buzz"
    }
    else if minutes % 3 == 0 {
        return "Fizz"
    }
    return "tick"
}

File: 7-kyu/fizz-buzz-cuckoo-clock/python/solution.py
def fizz_buzz_cuckoo_clock(time):
    time_split = time.split(':')
    hours = int(time_split[0])
    minutes = int(time_split[1])
    if minutes == 0:
        count = 12 if hours % 12 == 0 else hours - 12 if hours > 12 else hours
        return ' '.join(['Cuckoo' for _ in range(count)])
    elif minutes == 30:
        return 'Cuckoo'
    elif minutes % 15 == 0:
        return 'Fizz Buzz'
    elif minutes % 5 == 0:
        return 'Buzz'
    elif minutes % 3 == 0:
        return 'Fizz'
    return 'tick'